src/mds/MDSRank.cc

```diff
@@ -3847,3 +3873,31 @@ void MDSRank::send_task_status() {
 
   schedule_update_timer_task();
 }
+
+void MDSRank::schedule_inmemory_logger() {
+  dout(20) << __func__ << dendl;
+  timer.add_event_after(extraordinary_events_dump_interval,
+                        new LambdaContext([this]() {
+                          inmemory_logger();
+                        }));
+}
+
+void MDSRank::inmemory_logger() {
+  if (extraordinary_events_dump_interval &&
+      (client_eviction_dump || beacon.missed_beacon_ack_dump ||
+       beacon.missed_internal_heartbeat_dump || state < MDSMap::STATE_ACTIVE)) {
```

Comment:
BTW, why do we need to `or` the `state` here ?

Reply:
@lxbsz 
This is to dump the in-memory logs, if mds is in recovery state. Is there a better way to check if mds is in recovery state?

Reply:
I thought this PR will only dump recent logs when the `Log Level` is very low in all states if there is any important event happening. 

While your current code will dump the recent logs from the memory always periodically if state < STATE_ACTIVE even there has no any important event happened.

Maybe I missed something about the purpose of this fixing ?

Reply:
> I thought this PR will only dump recent logs when the `Log Level` is very low in all states if there is any important event happening.
> 
> While your current code will dump the recent logs from the memory always periodically if state < STATE_ACTIVE even there has no any important event happened.
> 

@lxbsz 
I'm inclined to dropping this requirement `mds is in a recovery state`. It comes from the tracker and it's different from the other events we have considered. You're right. This patch should look for the special events in all states, which is more important thing. Who knows how long the `recovery state` continues to pile up the logs by `dumping recent events`. So let's drop it.